nemo_aligner/utils/utils.py

```diff
@@ -123,7 +123,34 @@ def load_and_override_model_config(restore_path, model_cfg_to_overwrite, remove_
         checkpoint_cfg.pop("target", None)
         checkpoint_cfg.pop("nemo_version", None)
 
-    return OmegaConf.merge(checkpoint_cfg, model_cfg_to_overwrite)
+    resolved_checkpoint_cfg = resolve_overlapping_fields(checkpoint_cfg, model_cfg_to_overwrite)
+
+    return OmegaConf.merge(resolved_checkpoint_cfg, model_cfg_to_overwrite)
+
+
+def resolve_overlapping_fields(base_config, overwrite_config):
+    def recursive_merge(base, overwrite, path=[]):
+        for key, value in overwrite.items():
+            current_path = path + [key]
+
+            if key not in base:
+                continue
+
+            if isinstance(value, DictConfig) and isinstance(base[key], DictConfig):
+                recursive_merge(base[key], value, path=current_path)
+            else:
```

Comment:
We should check if `value` is True (one reason to make it a config is so we can easily set it to False if needed)

Reply:
Resolving as this is addressed in my suggested commit